### ISSUE TITLE
Cannot read apply of undefined (animateCamera)

### ISSUE THREAD

**Issue by chris-poe** [react-native-community/react-native-maps#2810]
### Is this a bug report?

Yes

### Have you read the [Installation Instructions](https://github.com/react-community/react-native-maps/blob/master/docs/installation.md)?

Yes

### Summary

My goal is to swap the dedicated camera methods with `animateCamera()`. Currently I am attepting to `center` the map given a lat/long. The old `animateToCoordinate` method is working as expected (see demo).

### Environment

Target iOS using: GoogleMaps

System:
      OS: Linux 4.9 Debian GNU/Linux 9 (stretch) 9 (stretch)
      Shell: 4.4.12 - /bin/bash
    Binaries:
      Node: 8.12.0 - /usr/local/bin/node
      Yarn: 1.12.3 - /usr/bin/yarn
      npm: 6.9.0 - /usr/local/bin/npm
      Watchman: 4.7.0 - /usr/local/bin/watchman
    npmPackages:
      expo: ^32.0.0 => 32.0.6
      react: 16.5.0 => 16.5.0
      react-native: 0.57.1
    npmGlobalPackages:
      expo-cli: 2.14.0

### Steps to Reproduce

Any attempt to call a partial object within `animateCamera()` with any of the doc specified keys will throw the error.

### Expected Behavior

Camera animates to specified lat/long.

### Actual Behavior
TypeError is thrown:

![Image from iOS](https://user-images.githubusercontent.com/25645289/56085949-6fd68080-5e1a-11e9-9793-bdcf5fb14046.png)

### Reproducible Demo

Expo snack was giving me issues so I created a repo:
https://github.com/chris-poe/rn-maps-issue-demo

**Comment by salah-ghanim** [comment]
@chris-poe componentDidMount gets called before render so of course you will run into crashes.

use onMapReady callback instead to animate to the coordinates

https://github.com/react-native-community/react-native-maps/blob/master/docs/mapview.md

**Comment by chris-poe** [comment]
@salah-ghanim unless I am misunderstanding, that does not resolve the issue:

    ...

      onMapReady = () => {
        this.map.animateCamera(
          {
            center: {
              latitude: 36.500805,
              longitude: -120.245361,
            }
          }
        );
      };

      render() {
        return (
          <MapView
            onMapReady={this.onMapReady}
            ref={r => this.map = r}
            provider="google"
            style={{ flex: 1 }}
          />
        );
      }

    ...

**Comment by chris-poe** [comment]
@salah-ghanim this was closed without resolution. The demo was updated and tested with your suggestion but the error is still occurring.

**Comment by rborn** [comment]
@chris-poe  I ran your code in a snack and expo seems to have an older version of the maps where the animateCamera is not present yet

```
 _this.map.animateCamera is not a function. (In '_this.map.animateCamera({
```
I also saw you tried to upgrade the module in package.json but as far as I know you cannot do this without ejecting from expo.

Try to replicate the issue with a non-expo demo and if it's still the case ping me or @salah-ghanim and we'll re-open the issue or make a new one

**Comment by chris-poe** [comment]
Thanks for the response @rborn. I was getting the same issue when trying to reproduce within a snack, so I created the repo linked under _reproducible demo_ section (https://github.com/chris-poe/rn-maps-issue-demo). That demo was made with `expo init` and I was able to install and display the most recent version and I can verify that the `animateCamera` method does exist in that environment. However, I'll attempt to reproduce in a none expo environment.
